### PR TITLE
Cel api update

### PR DESCRIPTION
**Pull request by thetechnick**
### Check List Before Merging

- [ ] This PR passes all pre-commit hook validations.
- [ ] This PR is fully tested and regression tests are included.
- [ ] Relevant documentation has been updated.

### CODE REVIEW

**Review by erdii — COMMENTED** [review]
Conceptually approving the API changes and ideas behind the PR. Preliminary review since the int test are not passing yet.

File: internal/packages/internal/packageimport/fs.go [hunk 1]
```diff
@@ -40,7 +96,17 @@ func FromFS(ctx context.Context, src fs.FS) (*packagetypes.RawPackage, error) {
 		case entry.IsDir():
 			// no special handling for directories
 
+		case entry.Type()&os.ModeSymlink == os.ModeSymlink:
+			// symlink
+			finalPath, err := src.Readlink(path)
+			if err != nil {
+				return err
+			}
+			_, err = os.Lstat(finalPath)
```
Comment (erdii) [inline internal/packages/internal/packageimport/fs.go:105]: this Lstat should not run against the host os but against the wrapped fs instance, correct?

File: internal/packages/internal/packageimport/fs_test.go [hunk 2]
```diff
@@ -3,46 +3,18 @@ package packageimport
...
-	rawPkg, err := FromFS(ctx, memFS)
-	require.NoError(t, err)
-	assert.Equal(t, validEntries, rawPkg.Files)
-}
-
 func TestFromFolder(t *testing.T) {
 	t.Parallel()
-	ctx := logr.NewContext(context.Background(), testr.New(t))
+	ctx := logr.NewContext(context.Background(), testr.NewWithOptions(t, testr.Options{
+		Verbosity: 99,
```
Comment (eqrx) [inline internal/packages/internal/packageimport/fs_test.go:16]: Why do you increase the verbosity here?

**Review by eqrx — CHANGES_REQUESTED** [review]

File: internal/packages/internal/packageimport/fs_test.go (hunk 2 above)
Comment (thetechnick) [reply internal/packages/internal/packageimport/fs_test.go:16]: Because verbosity 1 logs that are used within the business logic did not show. :)

**Review by eqrx — APPROVED** [review]

File: internal/packages/internal/packageimport/testdata/incl/test.yaml [hunk 3]
```diff
@@ -0,0 +1 @@
+xxx
```
Comment (eqrx) [inline internal/packages/internal/packageimport/testdata/incl/test.yaml:1]: no newline :(

**Review by eqrx — CHANGES_REQUESTED** [review]

**Review by eqrx — APPROVED** [review]

File: internal/packages/internal/packageimport/fs.go (hunk 1 above)
Comment (thetechnick) [reply internal/packages/internal/packageimport/fs.go]: no longer applies

File: internal/apis/manifests/packagemanifest_types.go [hunk 4]
```diff
@@ -67,41 +67,43 @@ type PackageManifestSpec struct {
...
+// PackageManifestPath is used to conditionally
 // render package objects based on their path.
-type PackageManifestConditionalPath struct {
+type PackageManifestPath struct {
 	// A file system path glob pattern.
 	// Syntax: https://pkg.go.dev/github.com/bmatcuk/doublestar@v1.3.4#Match
-	Glob string
+	Glob string `json:"glob"`
 	// A CEL expression with a boolean output type.
 	// Has access to the full template context and named conditions.
```
[the remark this answers is not in the document]
Comment (thetechnick) [reply internal/apis/manifests/packagemanifest_types.go:105]: The comments in the /internal/apis folder are unused and are not in sync with the public /apis folder.
I think we have to figure something out here to either strip comments automatically or keep them in sync. :thinking:

**Pull request merged**